src/components/main/MainCard.tsx

```diff
@@ -26,6 +28,32 @@ const MainCard: FC<MainCardProps> = ({ currentWeather, location }) => {
 
   const date = convertTime(time);
 
+  const getIsFavorite = (): boolean => {
+    const favoriteCities: LocationInfoType[] = getFavoriteCities();
+    return favoriteCities.some(city => city.id === location.id);
+  };
```

Comment:
function can be moved out of component body

Reply:
In function I use location object. If I move this function, then I need to send location in parameter. But I use this function, when defined "isFavorite" state. If I will call this function, when defined state (`getIsFavorite(location)`), then this function will call for every re-render. not only when componentDidMount.
I can move this function and update state isFavorite after mounting, but I think it excess render, which I can avoid